core/src/test/java/com/yahoo/imapnio/async/data/MessageNumberSetTest.java

```diff
@@ -144,6 +144,66 @@ public void testBuildStringWith0LengthMessageNumberSets() {
         Assert.assertNull(MessageNumberSet.buildString(new MessageNumberSet[0]), "Result mismatched.");
     }
 
+    /**
+     * Tests buildMessageNumberSets(String) method.
+     */
+    @Test
+    public void testBuildMessageNumberSetsWithOneString() throws ImapAsyncClientException {
+        final String msgSetStr = "1";
+        final MessageNumberSet[] messageNumberSets = MessageNumberSet.buildMessageNumberSets(msgSetStr);
+        Assert.assertNotNull(messageNumberSets, "Result mismatched.");
+        Assert.assertEquals(messageNumberSets.length, 1, "Result mismatched.");
+        Assert.assertEquals(MessageNumberSet.buildString(messageNumberSets), msgSetStr, "Result mismatched");
+    }
+
+    /**
+     * Tests buildMessageNumberSets(String) method.
+     */
+    @Test
+    public void testBuildMessageNumberSetsWithStartEnd() throws ImapAsyncClientException {
+        final String msgSetStr = "1:100";
+        final MessageNumberSet[] messageNumberSets = MessageNumberSet.buildMessageNumberSets(msgSetStr);
+        Assert.assertNotNull(messageNumberSets, "Result mismatched.");
+        Assert.assertEquals(messageNumberSets.length, 1, "Result mismatched.");
+        Assert.assertEquals(MessageNumberSet.buildString(messageNumberSets), msgSetStr, "Result mismatched");
```

Comment:
You are testing buildString() method here, but the test is to validate buildMessageNumberSets()
Can you test the values of messageNumberSets[0]?